src/reader.rs

```diff
@@ -737,6 +746,78 @@ where
         Ok(alias)
     }
 
+    fn read_schema_data(&mut self, attrs: HashMap<String, String>) -> Result<SchemaData, Error> {
+        let mut schema_data = SchemaData {
+            attrs,
+            ..Default::default()
+        };
+
+        loop {
+            let e = self.reader.read_event(&mut self.buf)?;
+            match e {
+                Event::Start(e) => match e.local_name() {
+                    b"SimpleData" => {
+                        let attrs = Self::read_attrs(e.attributes());
+                        if let Ok(simple_data) = self.read_simple_data(attrs) {
+                            schema_data.data.push(simple_data);
+                        }
+                    }
+                    b"SimpleArrayData" => {
+                        let attrs = Self::read_attrs(e.attributes());
+                        if let Ok(simple_array_data) = self.read_simple_array_data(attrs) {
+                            schema_data.arrays.push(simple_array_data);
+                        }
+                    }
+                    _ => {}
+                },
+                Event::End(e) => {
+                    if e.local_name() == b"SchemaData" {
+                        break;
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        Ok(schema_data)
+    }
+
+    fn read_simple_array_data(
+        &mut self,
+        attrs: HashMap<String, String>,
+    ) -> Result<SimpleArrayData, Error> {
+        let mut simple_array_data = SimpleArrayData {
+            attrs,
+            ..Default::default()
+        };
+
+        loop {
+            let e = self.reader.read_event(&mut self.buf)?;
+            match e {
+                Event::Start(e) => {
+                    if let b"value" = e.local_name() {
+                        simple_array_data.values.push(self.read_str()?);
+                    }
+                }
+                Event::End(e) => {
+                    if e.local_name() == b"SimpleArrayData" {
+                        break;
+                    }
+                }
+                _ => break,
+            }
+        }
+
+        Ok(simple_array_data)
+    }
+
+    fn read_simple_data(&mut self, attrs: HashMap<String, String>) -> Result<SimpleData, Error> {
+        Ok(SimpleData {
+            value: self.read_str()?,
+            attrs,
+        })
```

Comment:
It's a no-op for here now, but for consistency can we add `..Default::default()` here as well?

Reply:
Adding this triggers clippy's `needless_update` warning. I can put it in anyways if you'd like.